### ISSUE TITLE
Question: Support for mSATA SSD

### ISSUE THREAD

**Issue by chraaroe** [sakaki-/bcmrpi3-kernel-bis#7]
Hi sakaki, my apologies for using issues to ask a question.

I've been trying out your Gentoo build on my newly acqiured RPi 3 B+ for which I've bought one of [these](https://www.amazon.com/DIY-Pi-Desktop-Case-Raspberry/dp/B071G5RP41/ref=sr_1_3?keywords=raspberry+pi+desktop&qid=1557834979&s=gateway&sr=8-3). It comes with an extenion that allows an external mSATA SSD which for which I've purchased a Samsung 860.

However, upon trying it out I'm unable to find a way to get Gentoo to recognize the device. I see no trace of it through `dmesg` or anywhere else.

Would I need to compile my own kernel with some additional support to allow for drive to be recognized and mounted?

Thank you for your work.

**Comment by sakaki-** [comment]
Hi @chraaroe,
not sure about that specific drive, but if you can get it recognized when booted under stock Raspbian (with the same case, RPi3 B+ board etc.), and then identify the necessary kernel module(s) (following e.g. [this guide](https://unix.stackexchange.com/questions/248494/how-to-find-the-driver-module-associated-with-sata-device-on-linux)), I'll be happy to accept a PR to turn it/them on for this package (see e.g. [here](https://github.com/sakaki-/bcmrpi3-kernel-bis/pull/6) for an example PR).
Best, sakaki

**Comment by chraaroe** [comment]
It's a work in progress - I'm struggling a bit with trying to build my own kernel. Please allow me to use this occasion to thank you for your work and your thorough documentation. It's a pleasure to work with.

I'll see if I cant get it to work and report back my findings :)

**Comment by chraaroe** [comment]
Just an update on this one. I'm still investigating my issues with getting the kernel to recognize the harddrive. As recommended by you, I tried installing Raspbian on it to track down the driver used there. However, Raspbian didn't seem to identify the device either.

I'll look further into this. Could also just be a defective HAT.

**Comment by chraaroe** [comment]
I'll close this issue again. I never made it work on any distro and ended up concluding that it must have been a defect HAT. The mSATA drive itself was confirmed to work just fine.

After a bit of reading, I found that the controller on that HAT wasn't exactly praised. So instead of ordering a new identical HAT I settled for the X850 V3.0 mSATA SSD card instead. This was recognized from the get-go by Gentoo and has worked flawless so far.
